File: DAM 25-26/Acceso_a_datos/SwapShop/src/main/java/com/swapshop/modelos/Conversaciones.java
package com.swapshop.modelos;

import jakarta.persistence.*;
import lombok.*;

import java.sql.Timestamp;
import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "conversaciones")
public class Conversaciones {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    // Usuario que inicia o participa (1)
    @ManyToOne
    @JoinColumn(name = "id_usuario1", nullable = false)
    private Usuario usuario1;

    // Usuario que inicia o participa (2)
    @ManyToOne
    @JoinColumn(name = "id_usuario2", nullable = false)
    private Usuario usuario2;

    // Intercambio asociado
    @ManyToOne
    @JoinColumn(name = "id_intercambio", nullable = false)
    private IntercambiosPrestamos intercambio;

    @Column(name = "creado_en", columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
    private Timestamp creadoEn;

    // Relación bidireccional (opcional)
    @OneToMany(mappedBy = "conversacion", cascade = CascadeType.ALL)
    private List<Mensajes> mensajes;
}

File: DAM 25-26/DesarolloDeInterfaces/Front/SwapShop/src/app/modelos/Conversaciones.ts
import { Usuario } from './Usuario';
  import { IntercambioPrestamos } from './IntercambiosPrestamos';
  import { Mensaje } from './Mensajes';

  export interface ConversacionCrear {
    id_usuario1: number;
    id_usuario2: number;
    id_intercambio: number;
  }

  export interface Conversacion {
    id: number;
    id_usuario1: number;
    id_usuario2: number;
    id_intercambio: number;
    usuario1: Usuario;
    usuario2: Usuario;
    intercambio: IntercambioPrestamos;
    creadoEn: string; // ISO string
    mensajes: Mensaje[];
  }
